src/routerd_lib/handlers/proxy.cpp

```diff
@@ -45,88 +46,224 @@ namespace NAC {
+        std::cerr << "=== headers ===" << std::endl;
+
+        for (auto&& [header, values] : outgoing_request.Headers()) {
+            std::cerr << "  " << header << ":";
+
+            if (values.size() == 1) {
+                std::cerr << " " << values.at(0) << std::endl;
+
+            } else if (values.size() >= 2) {
+                for (auto&& value : values) {
+                    std::cerr << std::endl << "  " << value << std::endl;
+                }
+
+            } else {
+                std::cerr << " (empty)" << std::endl;
+            }
+        }
+        std::cerr << "=== end of headers ===" << std::endl;
+
+        std::cerr << "=== parts ===" << std::endl;
+
+        for (auto&& part : outgoing_request.Parts()) {
+            std::cerr << "[part]" << std::endl;
+            std::string ContentDisposition;
+            NHTTP::THeaderParams ContentDispositionParams;
+            NHTTPUtils::ParseHeader(
+                    part.Headers(),
+                    "content-disposition",
+                    ContentDisposition,
+                    ContentDispositionParams
+            );
+            std::cerr << "  content-length: " << part.ContentLength() << std::endl;
+            std::cerr << "  content-disposition: " << ContentDisposition << std::endl;
+            std::cerr << "  content-disposition-params: " << std::endl;
+
+            for (auto [key, value]: ContentDispositionParams) {
+                std::cerr << "    key='" << key << "', value='" << value << "'" << std::endl;
+            }
+
+            for (auto [name, value] : part.Headers()) {
+                std::cerr << "  [header] " << name << ": " << std::endl;
+
+                for (auto v : value) {
+                    std::cerr << "    " << v << std::endl;
+                }
+            }
+
+            std::cerr << "  [content]" << std::string(part.Content(), part.ContentLength())
+                      << "[/content]" << std::endl;
+            std::cerr << "[/part]" << std::endl;
+        }
+
+        std::cerr << "=== end of parts ===" << std::endl;
+        std::cerr << "== END OF OUTGOING REQUEST == " << std::endl;
+    }
+#endif
+
     void TRouterDProxyHandler::Iter(std::shared_ptr<TRouterDRequest> request, const std::vector<std::string>& args) const {
         auto&& graph = request->GetGraph();
 
         while (true) {
             bool somethingHappened(false);
             std::vector<std::string> failedServices;
 
+            // schedule next possible request
             for (auto&& treeIt : graph.Tree) {
                 if (!treeIt.second.empty() || request->IsInProgress(treeIt.first)) {
+                    // service has unprocessed dependency or is already being processed
+                    continue;
+                }
+
+                const auto& service = graph.Services.at(treeIt.first);
+
+                if (!service.SendRawOutputOf.empty() && !request->GetOutGoingRequest().PartByName(service.SendRawOutputOf)) {
```

Comment:
This case seems to be redundant here because `SendRawOutputOf` is already in dependencies and is checked prior to that line.